eng/scripts/Pack-Nuget.ps1

```diff
@@ -0,0 +1,156 @@
+	[string] $CommitSha,
+	[Parameter(Mandatory=$false)]
+	[string] $Branch
+)
+
+. "$PSScriptRoot/../common/scripts/common.ps1"
+$RepoRoot = $RepoRoot.Path.Replace('\', '/')
+
+$mcpServerjson = "$RepoRoot/eng/dnx/.mcp/server.json"
+$nuspecSourcePath = "$RepoRoot/eng/dnx/nuspec"
+$projectPropertiesScript = "$RepoRoot/eng/scripts/Get-ProjectProperties.ps1"
+
+if(!$ArtifactsPath) {
+	$ArtifactsPath = "$RepoRoot/.work/build"
+}
+
+if(!$OutputPath) {
+	$OutputPath = "$RepoRoot/.work/package"
+}
+
+if(!(Test-Path $ArtifactsPath)) {
+	LogError "Artifacts path $ArtifactsPath does not exist."
+	return
+}
+
+$sharedProjectProperties = & "$projectPropertiesScript" -ProjectName "Directory.Build.props"
+
+Push-Location $RepoRoot
+try {
+	# Clear and recreate the output directory
+	Remove-Item -Path $OutputPath -Recurse -Force -ErrorAction SilentlyContinue -ProgressAction SilentlyContinue
+
+	$wrapperJsonFiles = Get-ChildItem -Path $ArtifactsPath -Filter "wrapper.json" -Recurse
+	foreach($wrapperJsonFile in $wrapperJsonFiles) {
+		$serverDirectory = $wrapperJsonFile.Directory
+		$serverName = $serverDirectory.Name
+		$serverProjectProperties = & "$projectPropertiesScript" -ProjectName "$($serverName -replace '-native', '').csproj"
+		$platformOutputPath = "$OutputPath/nuget/$serverName/platform"
+		$wrapperOutputPath = "$OutputPath/nuget/$serverName/wrapper"
+
+		New-Item -ItemType Directory -Force -Path $platformOutputPath | Out-Null
+		New-Item -ItemType Directory -Force -Path $wrapperOutputPath | Out-Null
+
+		$wrapperPackageJson = Get-Content $wrapperJsonFile -Raw | ConvertFrom-Json -AsHashtable
+
+		$platformDirectories = Get-ChildItem -Path $serverDirectory -Directory
+
+		# Create dnx wrapper nuget tool
+        $tempNugetWrapperDir = Join-Path $wrapperOutputPath "temp"
+        $wrapperToolDir = "$tempNugetWrapperDir/tools/$($sharedProjectProperties.TargetFramework)/any"
+        $wrapperToolNuspec = "$tempNugetWrapperDir/$($serverProjectProperties.PackageId).nuspec"
+        New-Item -ItemType Directory -Force -Path $wrapperToolDir | Out-Null
+        New-Item -ItemType Directory -Force -Path "$tempNugetWrapperDir/.mcp" | Out-Null
+
+		$packageVersion = "$($serverProjectProperties.Version)$VersionSuffix"
+        
+        (Get-Content -Path "$nuspecSourcePath/RuntimeAgnosticToolSettingsTemplate.xml") `
+            -replace "__CommandName__", $serverProjectProperties.CliName |
+            Set-Content -Path "$wrapperToolDir/DotnetToolSettings.xml"
+
+        (Get-Content -Path $mcpServerjson -Raw) `
+            -replace '\$\(PackageDescription\)', $serverProjectProperties.Description `
+            -replace '\$\(PackageId\)', $serverProjectProperties.PackageId `
+            -replace '\$\(PackageVersion\)', $packageVersion `
+            -replace '\$\(RepositoryUrl\)', $RepoUrl |
+            Set-Content -Path "$tempNugetWrapperDir/.mcp/server.json"
+        
+        (Get-Content -Path "$nuspecSourcePath/RuntimeAgnosticTemplate.nuspec") `
```

Comment:
These templates look at little scary and might be a place for issues with changes. Can we please put some validation in place? Maybe something simple like ensuring there are no values (i.e. no "__" left in file) that are not replaced.